source/simulator/core.cc

```diff
@@ -1046,14 +1046,65 @@ namespace aspect
                                               face_coupling,
                                               Utilities::MPI::
                                               this_mpi_process(mpi_communicator));
+
+        if (solver_scheme_solves_advection_equations(parameters)
+            &&
+            compositional_fields_need_matrix_block(introspection))
+          {
+            // If we solve for more than one compositional field make sure we keep constrained entries
+            // to allow different boundary conditions for different fields.
```

Comment:
It took me a good long while to understand why you don't just merge the call to `make_flux_sparsity_pattern()` into the one above. Can you add a comment here or at the actual call site below mentioning that the point is that in the call here, "keeping constrained entries" implies passing `true` as fourth argument?